internal/controller/statusreport/statusreport_adapter.go

```diff
@@ -184,6 +186,50 @@ func (a *Adapter) EnsureSnapshotFinishedAllTests() (controller.OperationResult,
 	return controller.ContinueProcessing()
 }
 
+// EnsureGroupSnapshotCreationStatusReportedToGitProvider is an operation that will ensure the the group snapshot creation status is report to component snapshot
+func (a *Adapter) EnsureGroupSnapshotCreationStatusReportedToGitProvider() (controller.OperationResult, error) {
+	// Only report status for component Snapshots
+	if !gitops.IsComponentSnapshot(a.snapshot) || gitops.IsSnapshotCreatedByPACPushEvent(a.snapshot) {
+		return controller.ContinueProcessing()
+	}
+
+	if !metadata.HasAnnotation(a.snapshot, gitops.PRGroupCreationAnnotation) || !strings.Contains(a.snapshot.Annotations[gitops.PRGroupCreationAnnotation], gitops.FailedToCreateGroupSnapshotMsg) {
+		return controller.ContinueProcessing()
+	}
+
+	integrationTestStatus := intgteststat.GroupSnapshotCreationFailed
+
+	allIntegrationTestScenarios, err := a.loader.GetAllIntegrationTestScenariosForApplication(a.context, a.client, a.application)
+	if err != nil {
+		a.logger.Error(err, "Failed to get integration test scenarios for the following application",
+			"Application.Namespace", a.application.Namespace, "Application.Name", a.application.Name)
+		return controller.RequeueWithError(err)
+	}
+
+	if allIntegrationTestScenarios != nil {
+		tempGroupSnapshot := gitops.PrepareTempGroupSnapshot(a.application, a.snapshot)
+		filterIntegrationTestScenarios := gitops.FilterIntegrationTestScenariosWithContext(allIntegrationTestScenarios, tempGroupSnapshot)
```

Comment:
Does this mean that different context group snapshots won't be selected?

Reply:
What does this `different context group snapshots` mean?